paddle/fluid/operators/compare_op.h

```diff
@@ -67,7 +67,6 @@ class CompareOpKernel
     auto* x = context.Input<Tensor>("X");
     auto* y = context.Input<Tensor>("Y");
     auto* z = context.Output<Tensor>("Out");
-    z->mutable_data<T>(context.GetPlace());
```

Comment:
I am sorry that I don't understand why this line was there and why it should be removed. When I designed Tensor, the `mutable_data` method does reallocate the memory on the specified device. But I don't see why it would be a bug if we call `z->mutable_data()`, given that it should just try to allocate the memory.

Reply:
ElementwiseCOmputeEx does mutable_data for z.
Also the type is not correct here. It should be bool instead of T.

Reply:
Got it. Thank you very much for the fix!